zabbix_smartmontools/__init__.py

```diff
@@ -236,70 +288,198 @@ def getSmart(config, host, command, d):
 
         bandwidthCurrentRe = re.search(r'^SATA Version is:\s+.+current\:\s+(\d+\.\d+)\s+Gb\/s', p, re.M | re.I)
         if bandwidthCurrentRe:
             bandwidthCurrentResult = bandwidthCurrentRe.group(1)
         else:
             bandwidthCurrentResult = bandwidthMaxRe.group(1)
 
-        sender.append('%s smartctl.info[%s,bandwidthCurrent] "%s"' % (host, dR, bandwidthCurrentResult))
+        sender.append('%s smartctl.info[%s,bandwidthCurrent] "%s"' % (host, identifier, bandwidthCurrentResult))
 
     firmwareRe = re.search(r'^Firmware Version:\s+(.+)$', p, re.M | re.I)
     if firmwareRe:
-        sender.append('%s smartctl.info[%s,firmware] "%s"' % (host, dR, firmwareRe.group(1)))
+        sender.append('%s smartctl.info[%s,firmware] "%s"' % (host, identifier, firmwareRe.group(1)))
         driveHeader.append(firmwareRe.group(1))   # tracking duplicates
 
     vendorRe = re.search(r'^Vendor:\s+(.+)$', p, re.M | re.I)
     if vendorRe:
-        sender.append('%s smartctl.info[%s,vendor] "%s"' % (host, dR, vendorRe.group(1)))
+        sender.append('%s smartctl.info[%s,vendor] "%s"' % (host, identifier, vendorRe.group(1)))
 
     # SAS-only SSD value
     ssdwearRe = re.search(r'^Percentage used endurance indicator:\s+(\d+)', p, re.M | re.I)
     if ssdwearRe:
-        json.append({'{#DISKIDSSD}':dR})
-        sender.append('%s smartctl.value[%s,SSDwear] "%s"' % (host, dR, ssdwearRe.group(1)))
+        json.append({'{#DISKIDSSD}':identifier})
+        sender.append('%s smartctl.value[%s,SSDwear] "%s"' % (host, identifier, ssdwearRe.group(1)))
 
     # SATA values
     valuesRe = re.findall(r'^(?:\s+)?(\d+)\s+([\w-]+)\s+[\w-]+\s+\d{3}\s+[\w-]+\s+[\w-]+\s+[\w-]+\s+[\w-]+\s+[\w-]+\s+(\d+)', p, re.M | re.I)   # catch id, name and value
     if valuesRe:
-        sender.append('%s smartctl.info[%s,SmartStatus] "PRESENT_SATA"' % (host, dR))
+        sender.append('%s smartctl.info[%s,SmartStatus] "PRESENT_SATA"' % (host, identifier))
 
         for v in valuesRe:
-            json.append({'{#DVALUE%s}' % v[0]:dR, '{#SMARTNAME}':v[1]})
-            sender.append('%s smartctl.value[%s,%s] %s' % (host, dR, v[0], v[2]))
+            json.append({'{#DVALUE%s}' % v[0]:identifier, '{#SMARTNAME}':v[1]})
+            sender.append('%s smartctl.value[%s,%s] %s' % (host, identifier, v[0], v[2]))
 
     else:
-        getSmartSAS_Out = getSmartSAS(host, p, dR)
+        getSmartSAS_Out = getSmartSAS(host, p, identifier)
         if not getSmartSAS_Out[0]:
-            sender.append('%s smartctl.info[%s,SmartStatus] "PRESENT_SAS"' % (host, dR))
+            sender.append('%s smartctl.info[%s,SmartStatus] "PRESENT_SAS"' % (host, identifier))
             sender.extend(getSmartSAS_Out[1])
             json.extend(getSmartSAS_Out[2])
 
         else:
-            sender.extend(whyNoSmart(p, dR))
+            sender.extend(whyNoSmart(p, identifier))
+
+    return None, sender, json, (serial, driveHeader), identifier
```

Comment:
`json` was lost. Nothing but ID and STATUS are discovered.

Reply:
Could you elaborate?  I don't understand the problem.

Reply:
getSmart() adds many LLD items to be discovered to `json`. But, upon function return (getSmart_Out) json is not handled and therefore lost.